python/pyspark/context.py

```diff
@@ -864,8 +865,20 @@ def union(self, rdds):
         first_jrdd_deserializer = rdds[0]._jrdd_deserializer
         if any(x._jrdd_deserializer != first_jrdd_deserializer for x in rdds):
             rdds = [x._reserialize() for x in rdds]
-        cls = SparkContext._jvm.org.apache.spark.api.java.JavaRDD
-        jrdds = SparkContext._gateway.new_array(cls, len(rdds))
+        gw = SparkContext._gateway
+        jvm = SparkContext._jvm
+        jrdd_cls = jvm.org.apache.spark.api.java.JavaRDD
+        pair_jrdd_cls = jvm.org.apache.spark.api.java.JavaPairRDD
+        double_jrdd_cls = jvm.org.apache.spark.api.java.JavaDoubleRDD
+        if is_instance_of(gw, rdds[0]._jrdd, jrdd_cls):
+            cls = jrdd_cls
+        elif is_instance_of(gw, rdds[0]._jrdd, pair_jrdd_cls):
+            cls = pair_jrdd_cls
+        elif is_instance_of(gw, rdds[0]._jrdd, double_jrdd_cls):
+            cls = double_jrdd_cls
+        else:
+            raise TypeError("Unsupported java rdd class %s", rdds[0]._jrdd)
+        jrdds = gw.new_array(cls, len(rdds))
```

Comment:
Can you check https://github.com/apache/spark/pull/19498? The best way might be just add `lambda x: x` and make it `JavaRDD`.

Reply:
sure will take a look thanks

Reply:
@HyukjinKwon Addressed plz take a look again thanks